### ISSUE TITLE
Is 2d keypoint used or not?

### ISSUE THREAD

**Issue by xjwxjw** [zju3dv/GVHMR#32]
Hi, it seems that all data['kp2d'] are set with "torch.zeros(length, 17, 3)" used for training. 
![image](https://github.com/user-attachments/assets/5abf4dbf-2c4a-419c-bb20-1d38b21fb59e)

But in demo.py, data['kp2d'] is set torch.load(paths.vitpose). 
![image](https://github.com/user-attachments/assets/9d0b3b3f-97ad-4055-be7d-8947e8329dc0)

Is 2d keypoint used or not? Thanks.

**Comment by zehongs** [comment]
`kp2d` is the prediction of VitPose during inference, while during the training phase, it is obtained by augmenting the projection of kp3d

**Comment by xjwxjw** [comment]
If so, the kp2d is not aligned with the image feature, right? The image feature is fixed, but the kp2d is projected from augmented kp3d.

**Comment by zehongs** [comment]
Thanks for pointing out! I believe this should be removed for clarity, as batch['mask']['vitpose'] is always False.
